### PR TITLE
Improve onboarding, display server status and subscription status

### PR DESCRIPTION
**Pull request by 0x5378**
This PR contains:

* Onboarding rebuild
* Improved feedback when changes are saved
* API token authentication
* Server connection status displayed
* Billing status displayed
* General refactoring and clean up

### CODE REVIEW

**Comment by tonkapark** [comment]
Looks good.

File: includes/class-wc-taxjar-integration.php [hunk 1]
```diff
@@ -170,73 +179,141 @@ public function init_form_fields( ) {
...
+        'type'              => 'hidden',
+        'description'       => '<h3>Activate your TaxJar WooCommerce Plugin</h3>'
       ),
+
       'api_token' => array(
         'title'             => __( 'API Token', 'wc-taxjar' ),
         'type'              => 'text',
-        'description'       => __( '<a href="'.$this->app_uri.'account#api-access" target="_blank">Click here</a> to get your API token.', 'wc-taxjar' ),
-        'desc_tip'          => false,
+        'description'       => __( '<a href="'.$this->app_uri.'account#api-access" target="_blank">Click here</a> to get your API token.', 'wc-taxjar' ),        'desc_tip'          => false,
```
Comment (fastdivision) [inline includes/class-wc-taxjar-integration.php]: New line for `desc_tip`

File: includes/class-wc-taxjar-integration.php [hunk 2]
```diff
@@ -170,73 +179,141 @@ public function init_form_fields( ) {
...
+                        'Content-Type' => 'application/x-www-form-urlencoded'
+                      ),
+      'user-agent'  => $this->ua,
+      'body'        => $body_string
+    ) );
+
+    if ( ! is_wp_error( $response ) && $response['response']['code'] >= 200 && $response['response']['code'] < 300 ) {
+      $description_for_status .= '<div style="color: #3FAE2A;">';
+      $description_for_status .= __( 'Connection to TaxJar servers established.', 'wc-taxjar' );
+      $description_for_status .= '</>';
```
Comment (fastdivision) [inline includes/class-wc-taxjar-integration.php]: `</div>`

File: includes/class-wc-taxjar-integration.php [hunk 3]
```diff
@@ -170,73 +179,141 @@ public function init_form_fields( ) {
...
+      if( $this->post_or_setting('api_token') && isset( $body->valid ) && $body->valid === false ) {
+        $description_for_status .= '<br><br><div style="color: #ff0000;"><strong>';
+        $description_for_status .= 'It looks like your API token is invalid.';
+        $description_for_status .= sprintf( '<br><a href="%s" target="_blank">Please review your API token.</a>', $this->app_uri.'account#api-access' );
+        $description_for_status .='</strong></div>';
+        $api_valid = false;
+      }
+    } else {
+      $description_for_status .= '<div style="color: #ff0000;">';
+      $description_for_status .= __( 'wp_remote_post() failed. TaxJar could not connect to server. Please Contact your hosting provider.', 'wc-taxjar' );
```
Comment (fastdivision) [inline includes/class-wc-taxjar-integration.php]: Un-capitalize "Contact"

**Comment by fastdivision** [comment]
@0x5378 Noted some minor tweaks!

**Comment by 0x5378** [comment]
@fastdivision Thank you sir!

**Pull request merged**